Frontend/nutri-pal/src/app/diary/page.tsx

```diff
@@ -0,0 +1,223 @@
+        if (currentUser) {
+            setUser(currentUser);
+            setGetFlag(true)
+        }
+    })
+
+    // Do a GET request for the registered days that the user has with or without the query params
+    const getDates = (year = '', month = '', day = '') => {
+        const HISTORY_DAYS = BASE_URL_REST_API + 'history/registered_days/' + user.uid + `?year=${year}&month=${month}&day=${day}`;
+        axios.get(HISTORY_DAYS)
+            .then((response) => { setDates(response.data) })
+            .catch((error) => { alert('There has been an error trying to get the dates of the user \n' + error) })
+    }
+
+    // Do a GET request to get the maximum and minimum registered days the user has
+    const getLimitDates = () => {
+        const MAX_MIN_DATES = BASE_URL_REST_API + 'history/maxmindate/' + user.uid;
+        axios.get(MAX_MIN_DATES)
+            .then((response) => { setMaxDate(response.data._max.date.slice(0, 4)); setMinDate(response.data._min.date.slice(0, 4)) })
+            .catch((error) => { alert('There has been an error trying to get the maximum and minimum dates of the user \n' + error) })
+    }
+
+    // After auth make the GET requests to display the cards
+    useEffect(() => {
+        if (getFlag) {
+            getDates()
+            getLimitDates()
+        }
+    }, [getFlag])
+
+    // Function to open and closing the modal
+    const toggleModal = () => {
+        if (openModal) {
+            setOpenModal(false)
+        } else {
+            setOpenModal(true)
+        }
+    }
+
+    // When the input calendar date is changed then reset the values of the selected year and month filter, and then separate the string into year month and date for the GET request
+    useEffect(() => {
+        setFilterYear('')
+        setFilterMonth('')
+        if (filterDate.length > 0) {
+            const temp = filterDate.split('-');
+            getDates(temp[0], temp[1], temp[2]);
+        } else {
+            getDates()
+        }
+    }, [filterDate])
+
+    // When the a year is selected from the filter option then able the month filter option and do the GET request with the query params of just the year
+    useEffect(() => {
+        if (filterYear.length > 0) {
+            setMonthFlag(true)
+            getDates(filterYear)
+        } else {
+            setMonthFlag(false)
+            setFilterMonth('')
+            getDates()
+        }
+    }, [filterYear])
+
+    // When the month is selected from the filter option then do a GET request with they query params of the year and the month
+    useEffect(() => {
+        if (monthFlag) {
+            getDates(filterYear, filterMonth)
+        }
+    }, [filterMonth])
+
+    // When the minDate is received from the GET request when loading the page, then create a array to get all the years from the least to the greatest
+    useEffect(() => {
+        if (minDate.length > 0) {
+            const newRangeYear: any = []
```

Comment:
Heads up, out of scope for you, but ideally we can avoid using the "any" type as much as possible. I'm noticing a lot of any's. Typescript is not required so I don't think you should worry about it and focus more on writing lots of code. 

But just a heads up if you have time, if you use any during fulltime you will be asked to try as hard as you can to avoid it. 

Note: 
1. Flow is the type checker that meta uses for javascript
2. there is a flow version of Any